md/data/importer.py

```diff
@@ -1,56 +1,173 @@
+    blank = pd.DataFrame({'blank': ' '}, index=range(len(stops['STOPDATE'])))
+    stops['date'] = pd.to_datetime(stops['STOPDATE'].map(str) + blank['blank'].map(str) + stops['TIME_OF_STOP'].map(str))
+
+    stops['DOB_as_dt'] = pd.to_datetime(stops.DOB)
+    stops['computed_AGE'] = stops.apply(compute_AGE, axis=1)
+    stops.drop(['DOB_as_dt'], axis=1, inplace=True)
+
+    stops['index'] = range(1, len(stops) + 1)  # adds column at end
+    # move the index column to the front
+    stops = stops[stops.columns.tolist()[-1:] + stops.columns.tolist()[:-1]]
+    return stops
+
+
+def xls_to_csv(xls_path, csv_path):
+    assert not os.path.exists(csv_path)
+    stops = load_xls(xls_path)
+    stops = process_raw_data(stops)
+    logger.info("Converting {} > {}".format(xls_path, csv_path))
+    stops.to_csv(csv_path, index=False)
+    return stops
 
 
 def run(url, destination=None, download=True):
     """Download MD data, extract, convert to CSV, and load into PostgreSQL"""
     logger.info('*** MD Data Import Started ***')
-    download_and_unzip_data(url, destination)
+    destination = download_and_unzip_data(url, destination)
     # Convert to CSV
-    xls_path = os.path.join(destination, '2013.xlsx')
-    csv_path = os.path.join(destination, '2013.csv')
+    xls_path = get_datafile_path(url, destination)
+    csv_path = get_csv_path(url, destination)
     if not os.path.exists(csv_path):
-        logger.info('Loading {} into pandas'.format(xls_path))
-        df = pd.read_excel(xls_path, parse_dates={'datetime': ['Date of Stop']})
-        headers = ["Location",
-                   "Agency",
-                   "datetime",
-                   "Gender",
-                   "DOB",
-                   "Race",
-                   "County of Residence",
-                   "State of Residence",
-                   "State of Registration",
-                   "Stop Reason",
-                   "Search",
-                   "Search Reason",
-                   "Disposition",
-                   "Outcome"]
-        logger.info("Converting {} > {}".format(xls_path, csv_path))
-        df.to_csv(csv_path, columns=headers)
+        xls_to_csv(xls_path, csv_path)
     else:
         logger.info("{} exists, skipping XLS->CSV conversion".format(csv_path))
     csv_count = line_count(csv_path)
     logger.debug('Rows: {}'.format(csv_count))
     # use COPY to load CSV files as quickly as possible
-    copy_from(destination)
+    copy_from(csv_path)
 
 
-def copy_from(destination):
+def copy_from(csv_path):
     """Execute copy.sql to COPY csv data files into PostgreSQL database"""
     sql_file = os.path.join(os.path.dirname(__file__), 'copy.sql')
     cmd = ['psql',
-           '-v', 'data_dir={}'.format(os.path.abspath(destination)),
+           '-v', 'data_file={}'.format(csv_path),
            '-f', sql_file,
            settings.DATABASES['traffic_stops_md']['NAME']]
     if settings.DATABASES['traffic_stops_md']['USER']:
```

Comment:
This ``psql`` command will need to be updated to use the ETL user on staging/production since ``COPY`` requires superuser access. See the [NC](https://github.com/OpenDataPolicingNC/Traffic-Stops/blob/dev/nc/data/importer.py#L93-L94) code.